lang/semgrep-grammars/src/semgrep-kotlin/grammar.js

```diff
@@ -74,13 +74,41 @@ module.exports = grammar(standard_grammar, {
 	    );
 	},
 
+  // We would like to be able to parse programs which have a newline between the
+  // class name and the constructor:
+  // class Foo
+  // constructor Bar() { ... }
+
+  // The problem is that the Kotlin parser inserts a semicolon after "Foo", making
+  // it such that we get interrupted in the middle of the class_declaration.
+  // To make it so we can continue, we allow everything after the class identifier
+  // to be a standalone statement in its own right. This way, we can parse both parts
+  // individually, and stitch them together at parsing time.
+
+  // We only need to amend statements here, because the consumers of _declaration are
+  // only class_member_declaration, top_level_object and _statement.
+  // The former has `secondary_constructor`, which already looks like what we want to
+  // add, and the second seems to be unused.
+  // So we just need to fix _statement.
+  _statement: ($, previous) => choice(
+    previous,
+    prec.left(seq(
+      optional($.type_parameters),
+      seq(optional($.modifiers), "constructor"),
+      $._class_parameters,
+      optional(seq(":", $._delegation_specifiers)),
+      optional($.type_constraints),
+      optional($.class_body)
+    ))
```

Comment:
Very nice workaround. Thanks!

To simplify stitching during CST -> generic AST (which I'm currently working on using these changes), I think grouping this into its own rule such as `partial_class_declaration` would help.

Reply:
Wasn't sure if I should modify your PR directly or not. I created another PR that builds on top of this PR to refactor the rules a little bit because it makes it easier for the CST -> Generic AST process.

https://github.com/semgrep/ocaml-tree-sitter-semgrep/pull/477

Reply:
Sounds good, will do, thanks!

Reply:
Merged my commits into this branch.